File: apps/api/test/fixtures/multi-lang/payment.py
// @ts-nocheck
"""Payment processing module in Python."""
from .utils import Logger
from .gateway import PaymentGateway
from typing import Optional


class PaymentService:
    """Handles payment processing."""
    
    def __init__(self, api_key: str, timeout: int = 30):
        self.api_key = api_key
        self.timeout = timeout
        self.logger = Logger("PaymentService")
    
    def process_payment(self, amount: float, currency: str) -> bool:
        self.logger.info(f"Processing {amount} {currency}")
        gateway = PaymentGateway(self.api_key)
        return gateway.charge(amount, currency)
    
    def refund(self, transaction_id: str) -> None:
        self.logger.info(f"Refunding {transaction_id}")


def calculate_tax(amount: float, rate: float) -> float:
    return amount * rate


class PremiumPaymentService(PaymentService):
    """Premium tier with additional features."""
    
    def apply_discount(self, amount: float, discount: float) -> float:
        return amount * (1 - discount)

File: apps/api/test/fixtures/multi-lang/payment.php
// @ts-nocheck
<?php

namespace App\Payment;

use App\Utils\Logger;
use App\Gateway\PaymentGateway;

interface PaymentProcessor {
    public function processPayment(float $amount, string $currency): bool;
}

class PaymentService implements PaymentProcessor {
    private Logger $logger;
    private string $apiKey;

    public function __construct(string $apiKey, int $timeout = 30) {
        $this->apiKey = $apiKey;
        $this->logger = new Logger('PaymentService');
    }

    public function processPayment(float $amount, string $currency): bool {
        $this->logger->info("Processing {$amount} {$currency}");
        $gateway = new PaymentGateway($this->apiKey);
        return $gateway->charge($amount, $currency);
    }

    public function refund(string $transactionId): void {
        $this->logger->info("Refunding {$transactionId}");
    }
}

class PremiumPaymentService extends PaymentService {
    public function applyDiscount(float $amount, float $discount): float {
        return $amount * (1 - $discount);
    }
}

function calculateTax(float $amount, float $rate): float {
    return $amount * $rate;
}
